test/unit/BMath.tree

```diff
@@ -0,0 +1,149 @@
+BMathTest::calcSpotPrice
+├── when token weight in is zero
+│   └── it should revert // division by zero
+├── when token balance in too big
+│   └── it should revert // bi * BONE > uint256 max
```

Comment:
this happens inside the `bdiv`, should we test it here or only in BNum tests?
feels like adding it here involves dividing all trees into a lot of possible branches..